docs/FilePropertyManagement.md

```diff
@@ -2,37 +2,37 @@
+The zAppBuild framework supports overriding the majority of DBB build properties. The full list can be viewed at [application-conf/README.md](../samples/application-conf/README.md).
 
 zAppBuild leverages DBB's API and allows you to define build parameters on three different levels for each language script:
 
-  1. General defaults in corresponding the language property files - for example, defining the compile options for building COBOL programs in [application-conf/Cobol.properties](../samples/application-conf/Cobol.properties). Property keys make use of a language prefix, for instance for COBOL programs this is `cobol_`.
-  2. A group definition, to overriding the default by specifying using DBB's file property syntax using a pattern filter; or through a mapping to a language definition file to override the defaults, like [application-conf/languageDefinitionMapping.properties](../samples/MortgageApplication/application-conf/languageDefinitionMapping.properties). 
-  3. An individual file-level definition to override a single parameter leveraging Dependency Based Builds file properties syntax using [application-conf/file.properties](../samples/application-conf/file.properties) or multiple parameters for a specific file using the individual property file, like [epsmlist.cbl.properties](../samples/MortgageApplication/properties/epsmlist.cbl.properties).
+1. General defaults in the corresponding language property files: For example, you can define the compile options for building COBOL programs in [application-conf/Cobol.properties](../samples/application-conf/Cobol.properties). Property keys make use of a language prefix; for instance, COBOL property keys are prefixed with `cobol_`.
+2. A group definition that overrides the default in one of two ways:
+   - By using DBB's file property syntax and specifying the application artifact group via a pattern filter on the path name(s)
+   - By mapping to a language definition file to override the defaults, such as in [application-conf/languageDefinitionMapping.properties](../samples/MortgageApplication/application-conf/languageDefinitionMapping.properties)
+3. An individual file-level definition to override a single parameter via Dependency Based Build's file properties syntax using [application-conf/file.properties](../samples/application-conf/file.properties) or multiple parameters for a specific file using the individual property file. For example: [epsmlist.cbl.properties](../samples/MortgageApplication/properties/epsmlist.cbl.properties).
 
-zAppBuild comes with various strategies that can be combined via an order of precedence. The following table summarizes the strategies for overriding file properties from highest to lowest precedence:
+zAppBuild comes with various build property strategies that can be combined via an order of precedence. The following table summarizes the strategies for overriding file properties from highest to lowest precedence:
 
 ||Strategy|Use case|
```

Comment:
I think we should somehow explain how these Strategies are implemented ....


zAppBuild comes with various strategies that can be combined via an order of precedence. The following table summarizes the strategies for overriding file properties from highest to lowest precedence, including their underlying implementation strategy using the available DBB APIs:

||Precendence|Use case|Implementing strategy|
 |-|-|-|-|
|1.|Individual file properties|Override one or multiple build parameters for individual files|DBB file property defining an override for the specific file (*) |
|2.|Language definition mapping|Override and define one or multiple build parameters for a group of mapped application artifacts|DBB file property defining an override for the specific file(s) (*)|
|3.|DBB file properties|Override a single build parameter for individual files or a grouped list of application artifacts|DBB file property defining an override for the specific(*)  or grouped list(**) of application artifacts|
|4.|Default properties|General build properties used when no overrides are defined|Build property defining the default value for all files|

Notes:
(*) DBB is managing the DBB file properties in its separate internal table compared to the default properties. This table leverages the combination of [property name + file pattern] as the key of the internal table. When the same key is declared a second time, it overrides the first one. 
(**) Because of managing DBB file properties is done in a single table, you can experience unpredictable behaviour when mixing qualified file path pattern definitions and file path patterns containing wildcards for the same property name.

Reply:
I feel the implementation strategy is good information to include. However, I think it is a bit early in the document to present such fine details in the `(*)` and `(**)`. My perspective is that the main purpose of this section in the document is to introduce the different available override strategies and the use cases for them.

Perhaps we can keep the "Implementing strategy" column (although I would just name it "Implementation"), and move the `(*)` and `(**)` notes down to the bottom of the document as a [footnote](https://github.blog/changelog/2021-09-30-footnotes-now-supported-in-markdown-fields/)? That way, the emphasis remains on introducing/summarizing the strategies, but the implementation information is still available. What do you think of this?